Graph2 95% complete

PickleLily opened the pull request:
This has our graph 2 generation working. It is creating the graphs with the specialized output based on the files of interaction.


Issues ATM (mostly unimplemented):
- only uni-directional arrows (default to come FROM PID)
- Missing one conncetion? gets 2/3?
- graph is large? can we fix this?
- The same FD is being tracked as multiple nodes when it has different arguments?

PickleLily committed:
removing unneccesary folder

PickleLily committed:
deleted outdated folder

PickleLily committed:
removing more unneccesary files

PickleLily committed:
organizing

PickleLily committed:
organization and read me

PickleLily committed:
file parser working

PickleLily committed:
adding the three functionalities into one file...

PickleLily committed:
changed graph 1b to take in the unified falco input & restructured the data

PickleLily committed:
cleaned up repository further

PickleLily committed:
added automatic generation of PNG files...

PickleLily committed:
added falco output

PickleLily committed:
Merge branch 'unified_code' into lily/graph2

PickleLily committed:
new graph file parsing...

PickleLily committed:
Merge branch 'lilygraph2' of https://github.com/PickleLily/nginx_experimentation into lily/graph2

PickleLily committed:
added new graph

PickleLily committed:
fixed parsing errors

PickleLily committed:
added path shortening

PickleLily committed:
updated graph?????

PickleLily committed:
commiting the new graphs

PickleLily committed:
graph generated

PickleLily committed:
updated readme

PickleLily committed:
larger output

PickleLily committed:
updated with main

PickleLily committed:
...

PickleLily committed:
graph 2 output with parsed

PickleLily committed:
Merge branch 'ellabranch' into graph2

PickleLily committed:
started

PickleLily committed:
started ... real commit

PickleLily committed:
added edges and nodes

PickleLily committed:
finished add node & edge

PickleLily committed:
changes from w ella

DunneyBunny committed:
add_edge fixes - Ella

DunneyBunny committed:
find_or_add_node deference fix

DunneyBunny committed:
Error fixing plus my other work

DunneyBunny committed:
Merge branch 'graph2' of https://github.com/PickleLily/nginx_experimentation into graph2

DunneyBunny committed:
Added PID node to makeSubgraph

PickleLily committed:
fixed parsing error

DunneyBunny committed:
Sscanf fix

PickleLily committed:
rearranged the graph folder

DunneyBunny committed:
Merge branch 'graph2' of https://github.com/PickleLily/nginx_experimentation into graph2

DunneyBunny committed:
Update through makeSubgraph

DunneyBunny committed:
Merge branch 'graph2' of https://github.com/PickleLily/nginx_experimentation into graph2

DunneyBunny committed:
Updated formatting

PickleLily committed:
lily changes?

PickleLily committed:
combined lilys changes with ellas changes

PickleLily committed:
fixed parsing errors

PickleLily committed:
use the NodeID as the to and from for graphs

PickleLily committed:
creating multiple graphs WITH accept4 handled correctly!

PickleLily committed:
working graph2 generation --> wrong add node configuration (default to coming from PID node ONLY)

DunneyBunny committed:
Remidied issues ELLA was running into with printing graph info, POTENTIALLY WILL NEED TO ROLL BACK IF OTHER ISSUES PERSIST

DunneyBunny committed:
Changes did not go through because I had to update remote

DunneyBunny committed:
made h file

DunneyBunny committed:
More goofy Ella changes

DunneyBunny committed:
more ella changes

DunneyBunny committed:
Gives a "working solution" seemingly to our graph gen. Needs to be cleaned up.

DunneyBunny committed:
Took out railing printf statements that are unnecessary

DunneyBunny committed:
Sets currentFD of closed graphs to -1, so they can never be interacted with again by accident if FD gets reused

AprilZingher reviewed (commented):
I think this looks great! A lot of progress, and good work on the file parsing!

AprilZingher reviewed (approved):
Sorry I meant to make it an approve! These changes are great and a lot of progress!

PickleLily committed:
?

PickleLily committed:
Merge branch 'graph2' of https://github.com/PickleLily/nginx_experimentation into graph2

DunneyBunny committed:
Fixes?

DunneyBunny committed:
Re-organization of the project, alongside more clear naming conventions and READMEs.

Does not remedy the current issues existing within 'Final Graph Work', but makes project more clear in structure.
